### ISSUE TITLE
Confusing log when deploying dualStack

### ISSUE THREAD

**Issue by manuelbuil** [k3s-io/k3s#8797]
**Environmental Info:**
K3s Version: 

Node(s) CPU architecture, OS, and Version: 

Cluster Configuration:

**Describe the bug:**

When running dualStack you get the log:
```
level=info msg="The interface eth0 with ipv4 address 10.1.1.10 will be used by flannel"
```
Which can be confusing, especially if you are prioritizing IPv6 family.

**Steps To Reproduce:**

- Installed K3s: 

Deploy k3s in dualStack mode and check the logs

**Expected behavior:**

One log entry with both ip addresses

**Actual behavior:**

Two different log entries, one for IPv4 and one for IPv6

**Additional context / logs:**

**Comment by mdrahman-suse** [comment]
Pending fix: https://github.com/k3s-io/k3s/issues/8901
I was able to partially replicate and validate but need the above resolved to complete validation

**Comment by manuelbuil** [comment]
> Pending fix: #8901 I was able to partially replicate and validate but need the above resolved to complete validation

The log does not really depend on the external IP IPFamily or?

**Comment by mdrahman-suse** [comment]
### Validated on master branch with commit 96ebb963
 
## Environment Details

 
*Infrastructure*
- [ ] Cloud
- [ ] Hosted
 
*Node(s) CPU architecture, OS, and Version:*

```
Linux 5.19.0-1025-aws #26~22.04.1-Ubuntu SMP Mon Apr 24 01:58:15 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
PRETTY_NAME="Ubuntu 22.04.2 LTS"
VERSION_ID="22.04"
```
 
*Cluster Configuration:*

```
1 server
```
 
*Config.yaml:*

```
write-kubeconfig-mode: "0644"
tls-san:
  - fake.fqdn.value
cluster-init: true
node-name: ip-xxx-xxx-xx-244.us-west-1.compute.internal
cluster-cidr: 2001:cafe:42:0::/56,10.42.0.0/16
service-cidr: 2001:cafe:43:0::/112,10.43.0.0/16
token: <token>
node-ip: <IPv6>,<PRIVATE IPv4>
```
 
## Testing Steps

1. Copy config.yaml
```
$ sudo mkdir -p /etc/rancher/k3s && sudo cp config.yaml /etc/rancher/k3s
```
2. Install k3s
3. Check logs for the info
4. Should show both ipv4 and ipv6 IPs in the log in case of dual-stack
 
 
**Replication Results:**
- k3s version used for replication:

```
k3s version v1.28.3+k3s2 (bbafb86e)
go version go1.20.10
```
 

```
sudo journalctl -u k3s | grep -i interface
Nov 21 15:35:39 ip-xxx-xxx-xx-244 k3s[1895]: time="2023-11-21T15:35:39Z" level=info msg="The interface ens5 with ipv4 address <PRIVATE IPv4> will be used by flannel"
```
 
 
**Validation Results:**
- k3s version used for validation:

```
k3s version v1.28.3+k3s-96ebb963 (96ebb963)
go version go1.20.10
```
 

```
$ sudo journalctl -u k3s | grep -i interface
Nov 21 17:40:40 ip-xxx-xxx-xx-244 k3s[11438]: time="2023-11-21T17:40:40Z" level=info msg="Using dual-stack mode. The interface ens5 with ipv4 address <PRIVATE IPv4> and ipv6 address <IPv6> will be used by flannel"
```
